cmd/jwt.go

```diff
@@ -81,6 +82,34 @@ func authenticateJWTUsersWithCredentials(credentials auth.Credentials, expiresAt
 	return jwt.SignedString([]byte(serverCred.SecretKey))
 }
 
+// cachedAuthenticateNode will cache authenticateNode results for given values up to ttl.
+func cachedAuthenticateNode(ttl time.Duration) func(accessKey, secretKey, audience string) (string, error) {
+	type key struct {
+		accessKey, secretKey, audience string
+	}
+	type value struct {
+		created time.Time
+		res     string
+		err     error
+	}
+	cache, err := lru.NewARC(100)
+	if err != nil {
+		return authenticateNode
+	}
+	return func(accessKey, secretKey, audience string) (string, error) {
+		k := key{accessKey: accessKey, secretKey: secretKey, audience: audience}
+		v, ok := cache.Get(k)
+		if ok {
+			if val, ok := v.(*value); ok && time.Since(val.created) < ttl {
+				return val.res, err
+			}
+		}
+		s, err := authenticateNode(accessKey, secretKey, audience)
+		cache.Add(k, &value{created: time.Now(), res: s, err: err})
```

Comment:
It is not really going to change on next try since we are sending fixed parameters.

So we might as well cache it. Also `newCachedAuthToken` crashes the server on non-nil error, so it is largely academic. :D